stream/distributedlog/core/src/main/java/org/apache/distributedlog/BKAsyncLogReader.java

```diff
@@ -364,7 +364,9 @@ private boolean checkClosedOrInError(String operation) {
 
         Throwable cause = lastExceptionUpdater.get(this);
         if (null != cause) {
-            LOG.trace("Cancelling pending reads");
+            if (LOG.isTraceEnabled()) {
```

Comment:
this is not useful

Reply:
@nicoloboschi I think we should be consistent in style, consistent in form. What do you think?

Reply:
I think it's fine to leave the logging libraries to perform optimization if needed.
In this case there's no improvement.
Also, the code is less readable

It's fine to have cases where we check if some level is enabled to not perform values computation if not needed but in general it's not a good practice to guard every logging call